canonical/cloud-init
# Ubuntu/xenial

new-upstream-snapshot for sru into ubuntu/xenial

To create this branch:


```
cat > drop-deps.patch <<EOF
diff --git a/debian/changelog b/debian/changelog
index 61816397..926a2b0b 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -3,6 +3,8 @@ cloud-init (20.1-50-g7f9f33db-0ubuntu1~16.04.1) UNRELEASED; urgency=medium
   * d/control: drop pyflakes from Build-Depends
     - This python2 pyflakes package is an unneeded dependency now that tests
       use python3-pyflakes.
+  * d/control: drop python3-unittest2 and python3-nose from Build-Depends
+    - upstream now uses pytest
   * refresh patches:
    + debian/patches/ubuntu-advantage-revert-tip.patch
   * refresh patches:
diff --git a/debian/control b/debian/control
index 24328140..6f919475 100644
--- a/debian/control
+++ b/debian/control
@@ -14,7 +14,6 @@ Build-Depends: debhelper (>= 9),
                python3-jinja2,
                python3-jsonpatch,
                python3-mock,
-               python3-nose,
                python3-oauthlib,
                python3-pep8,
                python3-pyflakes,
@@ -23,7 +22,6 @@ Build-Depends: debhelper (>= 9),
                python3-serial,
                python3-setuptools,
                python3-six,
-               python3-unittest2,
                python3-yaml
 XS-Python-Version: all
 Vcs-Browser: https://git.launchpad.net/cloud-init/?h=ubuntu/devel
EOF

cat > manual-changelog.patch <<EOF
diff --git a/debian/changelog b/debian/changelog
index 7e127af6..ae78f242 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -3,6 +3,10 @@ cloud-init (20.2-38-g8377897b-0ubuntu1~16.04.1) UNRELEASED; urgency=medium
   * d/control: drop pyflakes from Build-Depends
     - This python2 pyflakes package is an unneeded dependency now that tests
       use python3-pyflakes.
+  * d/control: add python3-pytest to Build-Depends
+    - This fixes upstream daily builds.  python3-nose is not removed from
+      Build-Depends because, currently, the Ubuntu package builds will still
+      use it for testing.
   * d/control: drop python3-unittest2 and python3-nose from Build-Depends
     - upstream now uses pytest
   * refresh patches:
@@ -11,30 +15,6 @@ cloud-init (20.2-38-g8377897b-0ubuntu1~16.04.1) UNRELEASED; urgency=medium
    + debian/patches/ec2-classic-dont-reapply-networking.patch
    + debian/patches/openstack-no-network-config.patch
    + debian/patches/stable-release-no-jsonschema-dep.patch
-    - HACKING.rst: update CLA link (#199)
-    - Scaleway: Fix DatasourceScaleway to avoid backtrace (#128)
-      [Louis Bouchard]
-    - cloudinit/cmd/devel/net_convert.py: add missing space (#191)
-    - tools/run-container: drop support for python2 (#192) [Paride Legovini]
-    - Print ssh key fingerprints using sha256 hash (#188)
-    - Make the RPM build use Python 3 (#190) [Paride Legovini]
-    - cc_set_password: increase random pwlength from 9 to 20 (#189)
-    - .travis.yml: use correct Python version for xenial tests (#185)
-    - cloudinit: remove ImportError handling for mock imports (#182)
-    - Do not use fallocate in swap file creation on xfs. (#70) [Eduardo Otubo]
-    - .readthedocs.yaml: install cloud-init when building docs (#181)
-    - Introduce an RTD config file, and pin the Sphinx version to the RTD
-      default (#180)
-    - Drop most of the remaining use of six (#179)
-    - Start removing dependency on six (#178)
-    - Add Rootbox & HyperOne to list of cloud in README (#176) [Adam Dobrawy]
-    - docs: add proposed SRU testing procedure (#167)
-    - util: rename get_architecture to get_dpkg_architecture (#173)
-    - Ensure util.get_architecture() runs only once (#172)
-  * d/control: add python3-pytest to Build-Depends
-    - This fixes upstream daily builds.  python3-nose is not removed from
-      Build-Depends because, currently, the Ubuntu package builds will still
-      use it for testing.
   * New upstream snapshot. (LP: #1881018)
     - enable Puppet, Chef mcollective in default config (#385)
       [Mina Galić (deprecated: Igor Galić)]
@@ -203,6 +183,26 @@ cloud-init (20.2-38-g8377897b-0ubuntu1~16.04.1) UNRELEASED; urgency=medium
     - net/cmdline: correctly handle static ip= config (#201)
       [Dimitri John Ledkov]
     - Replace mock library with unittest.mock (#186)
+    - HACKING.rst: update CLA link (#199)
+    - Scaleway: Fix DatasourceScaleway to avoid backtrace (#128)
+      [Louis Bouchard]
+    - cloudinit/cmd/devel/net_convert.py: add missing space (#191)
+    - tools/run-container: drop support for python2 (#192) [Paride Legovini]
+    - Print ssh key fingerprints using sha256 hash (#188)
+    - Make the RPM build use Python 3 (#190) [Paride Legovini]
+    - cc_set_password: increase random pwlength from 9 to 20 (#189)
+    - .travis.yml: use correct Python version for xenial tests (#185)
+    - cloudinit: remove ImportError handling for mock imports (#182)
+    - Do not use fallocate in swap file creation on xfs. (#70) [Eduardo Otubo]
+    - .readthedocs.yaml: install cloud-init when building docs (#181)
+    - Introduce an RTD config file, and pin the Sphinx version to the RTD
+      default (#180)
+    - Drop most of the remaining use of six (#179)
+    - Start removing dependency on six (#178)
+    - Add Rootbox & HyperOne to list of cloud in README (#176) [Adam Dobrawy]
+    - docs: add proposed SRU testing procedure (#167)
+    - util: rename get_architecture to get_dpkg_architecture (#173)
+    - Ensure util.get_architecture() runs only once (#172)
 
  -- Chad Smith <chad.smith@canonical.com>  Thu, 28 May 2020 20:56:53 -0600

EOF

git fetch origin   # or upstream your git preference
git checkout origin/ubuntu/xenial -B ubuntu/xenial
patch -p1 < drop-deps.patch
git commit -m 'd/control: drop python3-nose and python3-unittest2' debian/control 
git commit -am 'changelog: add content for dropping python3-nose and python3-unittest2'

uss-tableflip/scripts/new-upstream-snapshot 8377897b
# manually replace SRU_BUG_NUMBER in the edit view of debian/changelog with 1881018
patch -p1 < manual-changelog.patch
git commit -m 'changelog: manual reorder content'
sed -i -e "1s/UNRELEASED/xenial/" debian/changelog
 git commit -m "releasing cloud-init version 20.2-38-g8377897b-0ubuntu1~16.04.1" debian/changelog
```

To test:
```
check diff of your branch vs mine (with out the patches)

run uss-tableflip/scripts/build-package
or make deb to ensure package builds
```
